components/support/utils/src/main/java/mozilla/components/support/utils/SafeUrl.kt

```diff
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.support.utils
+
+import android.content.Context
+
+/**
+ * Collection of methods used for ensuring the validity and security of URLs.
+ */
+object SafeUrl {
+    fun stripUnsafeUrlSchemes(context: Context, unsafeText: CharSequence?): String? {
+        val urlSchemesBlocklist = context.resources.getStringArray(R.array.mozac_url_schemes_blocklist)
+        var safeUrl = unsafeText.toString()
+        if (safeUrl.isEmpty()) {
+            return safeUrl
+        }
+
+        @Suppress("ControlFlowWithEmptyBody", "EmptyWhileBlock")
+        while (urlSchemesBlocklist.find {
+                if (safeUrl.startsWith(it, true)) {
```

Comment:
Thanks, I know it looks strange but this is what I get if I select "Reformat Code" in Android Studio.
It's because of the empty while block.
I could modify the indentation but then whenever somebody applies a general "Reformat Code" the indentation will be changed back to this.
So I would keep this as it is.